### PR TITLE
feat(react-component): add full feature component option

### PR DESCRIPTION
**Pull request by jmfrancois**
#**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

You can only create a component of one type in the available list

**What is the new behavior?**

in your app you may want to create a full feature component.
for this case you will create es6 class (using the generator you will have tests + styles)
but you will mix the three things in one:
* rendering HTML
* business logics using lifecyle and so on
* connect to the store to handle the state

If you do all that in one class you will have hard time to do great tests.

So the idea is you can create a component which is the results of 3 components:
* the stateless responsible for the rendering
* the stateful responsible for the feature
* the connect responsible to sync the state in redux

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:

```shell
     _-----_
    |       |    ╭──────────────────────────╮
    |--(o)--|    │ Welcome to the react-cmf │
   `---------´   │        generator!        │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |
   __'.___.'__
 ´   `  |° ´ Y `

? name MyComponent
? full component (pure + es6 + connect) Yes
? test snapshot
? css Yes
? path src/app/components
   create src/MyComponent/MyComponent.component.js
   create src/MyComponent/MyComponent.container.js
   create src/MyComponent/MyComponent.connect.js
   create src/MyComponent/MyComponent.test.js
   create src/MyComponent/index.js
   create src/MyComponent/MyComponent.scss
```

### CODE REVIEW

File: generators/react-component/index.js [hunk 1]
```diff
@@ -71,22 +76,37 @@ module.exports = yeoman.Base.extend({
...
+				this.destinationPath(`${this.props.path}/${this.props.name}/${this.props.name}.component.js`),
+				this
+			);
+			this.fs.copyTpl(
+				this.templatePath(`src/${this.props.test}.test.js`),
+				this.destinationPath(`${this.props.path}/${this.props.name}/${this.props.name}.test.js`),
+				this
+			);
+		} else {
+			['stateless', 'es6.class', 'connect'].forEach((t) => {
```
Comment (talend-glorieux) [inline generators/react-component/index.js]: I'm not a really big fan of that naming. What about 'component', 'container', 'connect'?
@Talend/frontend Tell us what you think ;)

Comment (jmfrancois) [inline generators/react-component/index.js]: for sure. It also can be 
'stateless', 'stateful', 'connect'

File: generators/react-component/index.js [hunk 2]
```diff
@@ -20,11 +20,19 @@ module.exports = yeoman.Base.extend({
 				return true;
 			},
 		}, {
+			type: 'confirm',
+			name: 'isfull',
```
Comment (frassinier) [inline generators/react-component/index.js]: Why it's not in camel case?

File: generators/react-component/templates/src/full.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+						<%= props.name %>: DEFAULT_STATE.toJS(),
+					},
+				}),
+			},
+		};
+		const props = mapStateToProps(state);
+		expect(typeof props).toBe('object');
+	});
+	it('should map state to props', () => {
+		function dispatch() {
```
Comment (frassinier) [inline generators/react-component/templates/src/full.test.js]: What's about `const dispatch = () => {};` ?

File: generators/react-component/templates/src/full.connect.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { connect } from 'react-redux';
+import Container, { DEFAULT_STATE } from './<%= props.name %>.container';
+
+export function getContainerInfo(ownProps) {
```
Comment (frassinier) [inline generators/react-component/templates/src/full.connect.js]: I don't understand the output, what is the role of `ownProps` here?

Comment (jmfrancois) [inline generators/react-component/templates/src/full.connect.js]: the ownprops is here to let you build the id from the props of the components.
So if for example you want to take the id props to identify the component in the state you will be able to do ownProps.id.
You can let it has is only if you know you will have only one instance of the component in the app.

**Pull request merged**